### PR TITLE
BOAC-2911, avoid permalink collisions on student profile page

### PR DESCRIPTION
**Pull request by johncrossman**
https://jira.ets.berkeley.edu/jira/browse/BOAC-2911

Also, resolves https://jira.ets.berkeley.edu/jira/browse/BOAC-2910

@paulkerschen - You introduced that regex. Do you know if this change will impact beyond note and appointment links in search results?

### CODE REVIEW

File: src/components/student/profile/AcademicTimeline.vue [hunk 1]
```diff
@@ -422,20 +422,21 @@ export default {
   },
   mounted() {
     if (this.anchor) {
-      const match = this.anchor.match(/#(\d[0-9A-Z-]+\d)/);
-      if (match) {
-        const messageId = match[1];
-        const note = this.find(this.messages, function(m) {
+      const match = this.anchor.match(/^#(\w+)-(\d+)/);
```
Comment (paulkerschen) [inline src/components/student/profile/AcademicTimeline.vue]: @johncrossman I think the legacy note ids already have hyphens in them, so this new permalink format might yield a fragment of the form `note-1234567-00001`, which I'm not sure this regex would catch.

Comment (paulkerschen) [reply src/components/student/profile/AcademicTimeline.vue]: Plus the ASC notes sometimes put letters in their note ids. You might need `this.anchor.match(/^#(\w+)-(\d[0-9A-Z-]+\d)/);` to catch all cases.

Comment (johncrossman) [reply src/components/student/profile/AcademicTimeline.vue]: @paulkerschen  Thank you. Forgot about those. But this latest regex would not catch one or two digit ids, right?

Comment (johncrossman) [reply src/components/student/profile/AcademicTimeline.vue]: I think I can tweak it right.

Comment (paulkerschen) [reply src/components/student/profile/AcademicTimeline.vue]: You're right, it wouldn't. Trust you to fix it.

**Comment by johncrossman** [comment]
Regex amended. cc: @paulkerschen

**Pull request merged**